### PR TITLE
[plugins/aws][feat] Add support for Cognito

### LINKED ISSUE #1175: [AWS] Support Cognito Service
### Motivation

Collect [AWS Cognito](https://aws.amazon.com/cognito) resources as part of the AWS collect cycle.

### Desired Behavior

Implement the resource collector as part of the [aws](https://github.com/someengineering/resoto/tree/main/plugins/aws) collector. To get an idea how a collector looks like, take the [SQS](https://github.com/someengineering/resoto/blob/main/plugins/aws/resoto_plugin_aws/resource/sqs.py) resource as inspiration. 

For further explanation and guidance, please visit the [Resource Guide](https://resoto.com/docs/edge/contributing/plugins/aws).

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by anjafr**
# Description

new resources:
- `aws_cognito_user_pool`
- `aws_cognito_user`
- `aws_cognito_group`

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Issues Fixed

- Fixes #1175 

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws/test/resources/files/apigateway/get-authorizers__2lsd9i45ub.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
             "name": "Authorization Authority",
             "type": "COGNITO_USER_POOLS",
             "providerARNs": [
-                "arn:aws:cognito-idp:eu-central-1:test:userpool/123"
+                "arn:aws:cognito-idp:us-east-1:123:userpool/123"
```
[the remark this answers is not in the document]
Comment (anjafr) [reply plugins/aws/test/resources/files/apigateway/get-authorizers__2lsd9i45ub.json:9]: That's so weird though. Everywhere I debugged into in the test run the region was `eu-central-1` 🤔

File: plugins/aws/resoto_plugin_aws/resource/cognito.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "delivery_medium": S("DeliveryMedium"),
+        "attribute_name": S("AttributeName"),
+    }
+    delivery_medium: Optional[str] = field(default=None)
+    attribute_name: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsCognitoUser(AwsResource):
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/cognito.py:66]: should also be a `BaseUser`

File: plugins/aws/resoto_plugin_aws/resource/cognito.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+    def delete_resource_tag(self, client: AwsClient, key: str) -> bool:
+        client.call(
+            aws_service="cognito-idp", action="untag-resource", result_name=None, ResourceArn=self.arn, TagKeys=[key]
+        )
+        return True
+
+    def delete_resource(self, client: AwsClient) -> bool:
+        client.call(aws_service="cognito-idp", action="delete-user-pool", result_name=None, UserPoolId=self.id)
+        return True
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/cognito.py:218]: I added the `called_mutator_apis` function. Please implement it here.

File: plugins/aws/resoto_plugin_aws/resource/apigateway.py [hunk 4]
```diff
@@ -211,8 +211,6 @@ class AwsApiGatewayAuthorizer(AwsResource):
     api_link: str = field(default=None)
 
     def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
-        # TODO add edge to Cognito User Pool when applicable (via self.authorizer_provider_arns)
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/apigateway.py:214]: 👍

**Review by aquamatthias — COMMENTED** [review]
Thanks. 2 small improvements - otherwise lgtm!

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Pull request merged**